model/labels.go

```diff
@@ -18,11 +18,74 @@
 package model
 
 import (
+	"encoding/json"
+	"fmt"
+	"strconv"
 	"strings"
 
 	"github.com/elastic/beats/v7/libbeat/common"
 )
 
+func filterLabels(labels common.MapStr, fn func(interface{}) interface{}) common.MapStr {
+	result := common.MapStr{}
+	for k, v := range labels {
+		if typeValue := fn(v); typeValue != nil {
+			if slice, ok := typeValue.([]interface{}); ok {
+				for i, elem := range slice {
+					result[fmt.Sprintf("%s_%d", k, i)] = elem
```

Comment:
Any particular reason why we don't just store an array of values? https://www.elastic.co/guide/en/elasticsearch/reference/current/array.html

Reply:
Not sure why, but I thought we were going to only allow a single level of key:value in the object, I'm not sure where I got that from.

Reply:
I think it would make sense to just index an array value with the specified field name, rather than exploding the labels. I'd say that will align better with user expectations.